java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -157,6 +176,46 @@ private static boolean generatesPrivateAccess(List<SymbolMetadata.AnnotationValu
     return values.stream().anyMatch(av -> "access".equals(av.name()) && "PRIVATE".equals(getAccessLevelValue(av.value())));
   }
 
+  private static boolean generatesPrivateAccess(ClassTree tree) {
+    if (usesAnnotation(tree, Collections.singletonList("lombok.Value"))) {
+      return true;
+    }
+    return Optional.ofNullable(tree.symbol().metadata().valuesForAnnotation("lombok.experimental.FieldDefaults"))
+      .map(List::stream)
+      .map(s -> s.filter(Objects::nonNull)
+        .anyMatch(av -> "level".equals(av.name()) && "PRIVATE".equals(getAccessLevelValue(av.value())))
+      ).orElse(false);
+  }
+
+  private static boolean generatesPrivateAccess(VariableTree tree) {
+    if (tree.parent() instanceof ClassTree && generatesPrivateAccess((ClassTree) tree.parent())) {
```

Comment:
Even if it would make sense that `tree.parent()` always return the same value, I believe it's safer to consider that it does not (to avoid tricky bugs).

Reply:
Fixed